Index: RTMath.cpp
===================================================================
--- RTMath.cpp	(revision 2676)
+++ RTMath.cpp	(working copy)
@@ -22,6 +22,9 @@
  ***************************************************************************/
 
 #include "RTMath.h"
+#if defined(__arm__)
+#include <time.h>
+#endif
 
 static float CentsToFreqTable[CONFIG_MAX_PITCH * 1200 * 2 + 1]; // +-1200 cents per octave
 
@@ -71,6 +74,11 @@
     return t;
     #elif defined(__APPLE__)
     return GetMachTime();
+	#elif defined(__arm__)
+    timespec tp;
+    clock_gettime(CLOCK_MONOTONIC, &tp);
+    return tp.tv_nsec;
+
     #else // we don't want to use a slow generic solution
     #  error "Sorry, LinuxSampler lacks time stamp code for your system."
     #  error "Please report this error and the CPU you are using to the LinuxSampler developers mailing list!"
